data_diff/diff_tables.py

```diff
@@ -183,52 +182,47 @@ def diff_tables(self, table1: TableSegment, table2: TableSegment) -> DiffResult:
         if self.bisection_factor < 2:
             raise ValueError("Must have at least two segments per iteration (i.e. bisection_factor >= 2)")
 
-        return self._diff_tables(table1, table2)
+        logger.info(
+            f"Diffing tables | segments: {self.bisection_factor}, bisection threshold: {self.bisection_threshold}."
+        )
 
-    def _diff_tables(self, table1, table2, level=0, segment_index=None, segment_count=None):
-        count1, count2 = self._precalc_attr_threaded("count", [table1, table2])
+        key_ranges = self._threaded_call("query_key_range", [table1, table2])
+        mins, maxs = zip(*key_ranges)
+
+        # We add 1 because our ranges are exclusive of the end (like in Python)
+        start_key = min(mins)
+        end_key = max(maxs) + 1
```

Comment:
I think the previous version where this is spelled is a bit more readable and explicit, but I don't feel too strongly.

Reply:
It feels easier to read for me. I guess it's subjective.

Also, this version is a preparation for more handling more than 2 tables.